cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -338,13 +338,13 @@ func (ks *kube2sky) generateSRVRecord(subdomain, portSegment, recordName, cName
 }
 
 func (ks *kube2sky) addDNS(subdomain string, service *kapi.Service) error {
-	if len(service.Spec.Ports) == 0 {
-		glog.Fatalf("Unexpected service with no ports: %v", service)
-	}
 	// if ClusterIP is not set, a DNS entry should not be created
 	if !kapi.IsServiceIPSet(service) {
 		return ks.newHeadlessService(subdomain, service)
 	}
+	if len(service.Spec.Ports) == 0 {
+		glog.Fatalf("Unexpected service with no ports: %v", service)
```

Comment:
I concur with Tim's observation..
We can just delete that line..
In the generateRecordsForPortalService call, we can add a simple log.info to say that srv records wont be created since there are no ports declared...